### PR TITLE
Describe workflow intefrace

### PR DESCRIPTION
**Pull request by EmilyGavrilenko**
# Description

Adding capability to describe workflow interface - both inputs and outputs regarding types of data:

The following workflow:
```
{
    "version": "1.0",
    "inputs": [
        {"type": "WorkflowImage", "name": "image"},
        {"type": "WorkflowParameter", "name": "model_id"},
    ],
    "steps": [
        {
            "type": "ObjectDetectionModel",
            "name": "general_detection",
            "image": "$inputs.image",
            "model_id": "$inputs.model_id",
            "class_filter": ["dog"],
        },
        {
            "type": "Crop",
            "name": "cropping",
            "image": "$inputs.image",
            "predictions": "$steps.general_detection.predictions",
        },
        {
            "type": "ClassificationModel",
            "name": "breds_classification",
            "image": "$steps.cropping.crops",
            "model_id": "dog-breed-xpaq6/1",
        },
    ],
    "outputs": [
        {
            "type": "JsonField",
            "name": "detections",
            "selector": "$steps.general_detection.predictions",
        },
        {
            "type": "JsonField",
            "name": "crops",
            "selector": "$steps.cropping.crops",
        },
        {
            "type": "JsonField",
            "name": "classification",
            "selector": "$steps.breds_classification.*",
        },
    ],
}
```

will return:
```json
{
    "inputs": {
        "image": ["image"],
        "model_id": ["roboflow_model_id"],
    },
    "outputs": {
        "detections": ["object_detection_prediction"],
         "crops": ["image"],
         "classification": {
            "inference_id": ["string"],
            "predictions": ["classification_prediction"],
         },
    },
    "typing_hints": {
        "image": "dict",
        "roboflow_model_id": "str",
        "object_detection_prediction": "dict",
        "string": "str",
        "classification_prediction": "dict",
    },
    "kinds_schemas": {
        "image": {},
        "object_detection_prediction": {"dict": "with OpenAPI 3.0 schema of result"},
        "classification_prediction": {"dict": "with OpenAPI 3.0 schema of result"}
    }
}

```

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

* CI still 🟢 
* new tests
* e2e tests

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

**Review by PawelPeczek-Roboflow — COMMENTED** [review]
the same can be achieved having schemas of blocks on the UI end - why secondary request to backend to run post-processing of schemas needed?

**Comment by EmilyGavrilenko** [comment]
> the same can be achieved having schemas of blocks on the UI end - why secondary request to backend to run post-processing of schemas needed?

Leo from Pella requested an API endpoint to call to get the schema for the workflow response ([slack thread](https://roboflow.slack.com/archives/C06BD36S37A/p1725646269857019?thread_ts=1725645936.014469&cid=C06BD36S37A)). Will be used for integration with their downstream pipelines

**Comment by PawelPeczek-Roboflow** [comment]
ok then - let's mark it to be released in `0.19.0`

**Comment by PawelPeczek-Roboflow** [comment]
taking that and preparing for rel

File: inference/core/interfaces/http/http_api.py [hunk 1]
```diff
@@ -956,6 +960,41 @@ async def infer_lmm(
 
         if not DISABLE_WORKFLOW_ENDPOINTS:
 
+            @app.post(
```
Comment (PawelPeczek-Roboflow) [inline inference/core/interfaces/http/http_api.py:963]: that needs to be added into hosted API routing

Comment (hansent) [reply inference/core/interfaces/http/http_api.py:979]: I think we need to add it here like the other ones:  https://github.com/roboflow/roboflow-infra/blob/1e58b097dc081a18851588a253d18ac3e4539b1c/aws/serverless/modules/containerlambda/main.tf#L128-L144

Comment (hansent) [reply inference/core/interfaces/http/http_api.py:979]: @EmilyGavrilenko is the customer requesting this running on hosted API or dedicates/local?

Comment (PawelPeczek-Roboflow) [reply inference/core/interfaces/http/http_api.py:979]: that's hosted most likely given Nick comment
I posted this comment not to forget

Comment (PawelPeczek-Roboflow) [reply inference/core/interfaces/http/http_api.py:979]: regardless - this is potentially useful feature for preview in UI

Comment (hansent) [reply inference/core/interfaces/http/http_api.py:979]: infra PR here:  https://github.com/roboflow/roboflow-infra/pull/334

**Comment by PawelPeczek-Roboflow** [comment]
Added `typing_hints` and `kinds_schemas`. Not in the way I like that - would require re-iteration to ease maintenance, but works for now and is added aside of anything else, not requesting drastic changes.

`kinds_schemas` provide OpenAPI 3.0 schemas for entities, but obviously not for all, simple type hints for kinds are delivered under`typing_hints` key

**Review by hansent — DISMISSED** [review]
lgtm tested locally

we only need this for explicit spec?  i.e. not able to get it for name workflow via workflow url?

**Comment by hansent** [comment]
> `we only need this for explicit spec? i.e. not able to get it for name workflow via workflow url?` - not sure if I understand

like I have to post the workflow spec, but I can't post hte workflow url and have it looked up via API like I can for running it?

**Comment by PawelPeczek-Roboflow** [comment]
still do not understand - I feel like both are possible

**Review by hansent — APPROVED** [review]
reapproving after fixing block names

**Pull request merged**